src/aks-preview/azext_aks_preview/maintenanceconfiguration.py

```diff
@@ -7,47 +7,170 @@
-        timeInWeek = TimeInWeek(**dict)
-        result = MaintenanceConfiguration()
-        result.time_in_week = [timeInWeek]
-        result.not_allowed_time = []
-        return result
-
-    maintenance_config = get_file_json(config_file)
-    logger.info(maintenance_config)
-    return maintenance_config
-
-
-def aks_maintenanceconfiguration_update_internal(
-    cmd,
-    client,
-    resource_group_name,
-    cluster_name,
-    config_name,
-    config_file,
-    weekday,
-    start_hour
-):
-    logger.info('resource_group_name: %s, cluster_name: %s, config_name: %s, config_file: %s, weekday: %s, start_hour: %s ', resource_group_name, cluster_name, config_name, config_file, weekday, start_hour)
-
-    config = getMaintenanceConfiguration(cmd, config_file, weekday, start_hour)
-    return client.create_or_update(resource_group_name=resource_group_name, resource_name=cluster_name, config_name=config_name, parameters=config)
+def constructDedicatedMaintenanceConfiguration(cmd, raw_parameters):
+    weekday = raw_parameters.get("weekday")
+    start_hour = raw_parameters.get("start_hour")
+    if weekday is not None or start_hour is not None:
+        raise MutuallyExclusiveArgumentError('--weekday and --start-hour are only supported for default maintenance configuration.')
+    
+    maintenanceConfiguration = cmd.get_models('MaintenanceConfiguration', resource_type=CUSTOM_MGMT_AKS_PREVIEW, operation_group='maintenance_configurations') 
+    result = maintenanceConfiguration()
+    result.maintenance_window = constructMaintenanceWindow(cmd, raw_parameters)
+    return result
+
+def constructMaintenanceWindow(cmd, raw_parameters):
+    schedule = constructSchedule(cmd, raw_parameters)
+    start_date = raw_parameters.get("start_date")
+    start_time = raw_parameters.get("start_time")
+    duration_hours = raw_parameters.get("duration_hours")
+    utc_offset = raw_parameters.get("utc_offset")
+
+    MaintenanceWindow = cmd.get_models('MaintenanceWindow', resource_type=CUSTOM_MGMT_AKS_PREVIEW, operation_group='maintenance_configurations')
+    maintenanceWindow = MaintenanceWindow(
+        schedule=schedule,
+        start_date=start_date,
+        start_time=start_time,
+        duration_hours=duration_hours,
+        utc_offset=utc_offset
+    )
+    return maintenanceWindow
+
+def constructSchedule(cmd, raw_parameters):
+    schedule_type = raw_parameters.get("schedule_type")
+    interval_days = raw_parameters.get("interval_days")
+    interval_weeks = raw_parameters.get("interval_weeks")
+    interval_months = raw_parameters.get("interval_months")
+    day_of_week = raw_parameters.get("day_of_week")
+    day_of_month = raw_parameters.get("day_of_month")
+    week_index = raw_parameters.get("week_index")
+    
+    Schedule = cmd.get_models('Schedule', resource_type=CUSTOM_MGMT_AKS_PREVIEW, operation_group='maintenance_configurations')
+    schedule = Schedule()
+
+    if schedule_type == CONST_DAILY_MAINTENANCE_SCHEDULE:
```

Comment:
Wow, so many new if-else, please add some UT to cover as many of them as possible

Reply:
yeah sorry I know it's hard to review :(  I added a new test file `test_maintenanceconfiguration.py` for those UTs